server/src/middleware/recipeValidator.ts

```diff
@@ -0,0 +1,34 @@
+import Joi, { ValidationResult, string } from "joi";
+import { Request, Response, NextFunction } from "express";
+// import { ObjectId } from "mongodb";
+
+export function recipeBodyValidator(schema: Joi.ObjectSchema) {
+  return (req: Request, res: Response, next: NextFunction) => {
+    const { error } = schema.validate(req.body);
+
+    if (!error) {
+      return next();
+    }
+
+    const { details } = error;
+    const message = details.map((i) => i.message).join(",");
+
+    console.log("error", message);
+    res.status(422).json({ error: message });
+  };
+}
+
+export function recipeIdValidator(schema: Joi.StringSchema) {
```

Comment:
I'd say this validator is too specific. In future you might have routes with more than one path param and you won't be able to use this validator for them, because this one looks only for `id` path param. I suggest you to create a validation schema that also accepts an object and pass whole `req.params` object to it.  In that the validator will be more extensible and reusable.